homeassistant/components/ovo_energy/sensor.py

```diff
@@ -0,0 +1,212 @@
+"""Support for OVO Energy sensors."""
+from datetime import datetime, timedelta
+import logging
+
+import aiohttp
+from ovoenergy import OVODailyElectricity, OVODailyGas, OVODailyUsage
+from ovoenergy.ovoenergy import OVOEnergy
+
+from homeassistant.components.ovo_energy import OVOEnergyDeviceEntity
+from homeassistant.components.ovo_energy.const import (
+    DATA_CLIENT,
+    DATA_COORDINATOR,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=300)
+PARALLEL_UPDATES = 4
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up OVO Energy sensor based on a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+    client: OVOEnergy = hass.data[DOMAIN][entry.entry_id][DATA_CLIENT]
+
+    now = datetime.utcnow()
+    try:
+        usage: OVODailyUsage = await client.get_daily_usage(now.strftime("%Y-%m"))
+        currency = usage.electricity[len(usage.electricity) - 1].cost.currency_unit
+    except aiohttp.ClientError as exception:
+        _LOGGER.warning(exception)
+        raise PlatformNotReady from exception
```

Comment:
This data is already available in the `coordinator` at this point, instead of querying it again, you can just use `coordinator.data.electricity...`